src/main/java/eu/xenit/alfred/content/gateway/routing/ServiceTracker.java

```diff
@@ -0,0 +1,61 @@
+package eu.xenit.alfred.content.gateway.routing;
+
+import eu.xenit.alfred.content.gateway.servicediscovery.AppService;
+import eu.xenit.alfred.content.gateway.servicediscovery.ServiceAddedHandler;
+import eu.xenit.alfred.content.gateway.servicediscovery.ServiceDeletedHandler;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.gateway.event.RefreshRoutesEvent;
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder;
+import org.springframework.context.ApplicationEventPublisher;
+import reactor.core.publisher.Flux;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ServiceTracker implements ServiceAddedHandler, ServiceDeletedHandler, RouteLocator {
+    private final ApplicationEventPublisher publisher;
+    private final RouteLocatorBuilder routeLocatorBuilder;
+
+    private final Map<String, AppService> services = new ConcurrentHashMap<>();
```

Comment:
the key here is the applicaiton-id, which will fails if we have multiple deployments running